### ISSUE TITLE
The Logs in Travis don't colour code class is Undeclared errors

### ISSUE THREAD

**Issue by macta** [hpi-swa/smalltalkCI#298]
It's confusing that a possible error like an undeclared class (due to a bad package prereq baseline of) isn't very visible in the logs.

e.g.
Fetched -> Lambda-cypress.1 --- filetree:///home/travis/build/macta/PharoLambda [:] --- filetree:///home/travis/build/macta/PharoLambda
Lambda class>>processJSON: (NeoJSONReader is Undeclared) 

Loaded -> Lambda-cypress.1 --- filetree:///home/travis/build/macta/PharoLambda [:] --- cache
...finished baseline

The third line should be red.

As an aside - this line didn't even appear in the non-collapsed log - not sure why Travis chooses to hide it? (it did show a later undeclared error for a smalltalkCI doit however, and that wasn't red either).

I suspect this would require the smalltalk logger to be doctored somehow.

**Comment by fniephaus** [comment]
Yes, errors are currently not colored. I'm sure there's a way to do it...

**Comment by peteruhnak** [comment]
> It's confusing that a possible **error** like an undeclared class

it is not an error, it is a warning; and in most cases it is just noise, as the loader can deal with missing classes that are added later during loading. I don't think Travis/smalltalkCI can easily make automated decision whether the loading have actually failed or not.

Maybe playing with Baseline loading order could suppress some if it, but I don't think people are paying attention to the load info unless something have failed.

In the cases that a dependency is actually missing (I regularly forget to add dependencies), I will know that it is missing, because my tests have failed.

**Comment by macta** [comment]
They should at least colour yellow.

Honestly, I think you should sort load orders out and load clean unless you toggle some flag (like having no tests does).

But I guess it's a personal preference.

When I was at OTI (and later st JPMorgan ) this kind of warning was considered an error and you were asked to fix it.

But I'm newish to the current ethos

**Comment by fniephaus** [comment]
The message is printed as part of `UndeclaredVariableWarning`'s defaultAction when Metacello loads a project. I think it'd make more sense to add an option to Metacello to fail on these kind of warnings. Maybe open an issue at https://github.com/dalehenrich/metacello-work/issues?

**Comment by macta** [comment]
Ok - I've submitted 2 issues to Metacello - 1 for an option to conditionally fail on such errors (if you had this ability, it could be a flag for Smalltalk CI - I guess set to false for the reasons Peter documents, but for me I would prefer to run in a stricter manner). The second is an option to customise errors from Metacello so you add ansi colour coding as a prefix to them for better output in travis.

As there hasn't been much activity from what I can see, it probably won't happen - but I may investigate myself.

**Comment by fniephaus** [comment]
Ok, turns out we can do something about this in smalltalkCI. Oddly enough, `OCSemanticWarning`s do not inherit from `Warning`s and are not handled [here](https://github.com/hpi-swa/smalltalkCI/blob/3f51d6c4c3e523e603d6216c0fd3fb7fcfae91d9/repository/SmalltalkCI-Core.package/SCIMetacelloLoadSpec.class/instance/basicLoadProjectOn..st#L16-L24).
I hope to have a fix soon...
